packages/editor/src/components/post-author/index.js

```diff
@@ -1,67 +1,46 @@
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
-import { withInstanceId, compose } from '@wordpress/compose';
-import { Component } from '@wordpress/element';
-import { withSelect, withDispatch } from '@wordpress/data';
+import { __experimentalReadSelect, useDispatch } from '@wordpress/data';
 import { decodeEntities } from '@wordpress/html-entities';
+import { CustomSelect, Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { Suspense } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import PostAuthorCheck from './check';
 
-export class PostAuthor extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.setAuthorId = this.setAuthorId.bind( this );
-	}
-
-	setAuthorId( event ) {
-		const { onUpdateAuthor } = this.props;
-		const { value } = event.target;
-		onUpdateAuthor( Number( value ) );
-	}
-
-	render() {
-		const { postAuthor, instanceId, authors } = this.props;
-		const selectId = 'post-author-selector-' + instanceId;
-
-		// Disable reason: A select with an onchange throws a warning
-
-		/* eslint-disable jsx-a11y/no-onchange */
-		return (
-			<PostAuthorCheck>
-				<label htmlFor={ selectId }>{ __( 'Author' ) }</label>
-				<select
-					id={ selectId }
-					value={ postAuthor }
-					onChange={ this.setAuthorId }
-					className="editor-post-author__select"
-				>
-					{ authors.map( ( author ) => (
-						<option key={ author.id } value={ author.id }>{ decodeEntities( author.name ) }</option>
-					) ) }
-				</select>
-			</PostAuthorCheck>
-		);
-		/* eslint-enable jsx-a11y/no-onchange */
-	}
-}
-
-export default compose( [
-	withSelect( ( select ) => {
-		return {
-			postAuthor: select( 'core/editor' ).getEditedPostAttribute( 'author' ),
-			authors: select( 'core' ).getAuthors(),
-		};
-	} ),
-	withDispatch( ( dispatch ) => ( {
-		onUpdateAuthor( author ) {
-			dispatch( 'core/editor' ).editPost( { author } );
-		},
-	} ) ),
-	withInstanceId,
-] )( PostAuthor );
+const authorId = __experimentalReadSelect( 'core/editor' ).getEditedPostAttribute(
+	'author'
+);
+const authors = __experimentalReadSelect( 'core' ).getAuthors();
```

Comment:
This reads from the default registry and doesn't honor the `RegistryProvidder` that might have been applied. How can we solve this without complexity?

Reply:
You can't really, without enforcing that this should always be called, so that it can read from context using `useContext`, and that kind of breaks a lot of the flexibility this API gives you for pre-fetching data in parallel to view/code loading.

What we can do is additionally provide a hook that returns `__experimentalReadSelect` from the nearest `RegistryProvider`:

```js
const __experimentalReadSelect = __experimentalUseReadSelect();
```

That would solve for this without adding complexity to the most common use case which is reading from the default registry.